### ISSUE TITLE
How to decide the hyperparameter of dataset preprocessing

### ISSUE THREAD

**Issue by 22TonyFStark** [med-air/3DSAM-adapter#7]
Hello, I noticed that different hyperparameters are used for different datasets, I wonder how these hyperparameters are chosen(or maybe calculated), could you give some tips? Thanks~
The primary hyperparameters include Intensity Clipping Range, Global Foreground Voxel Intensity Mean/Std, Patch Size, and do_dummy_2D.

**Comment by peterant330** [comment]
> Hello, I noticed that different hyperparameters are used for different datasets, I wonder how these hyperparameters are chosen(or maybe calculated), could you give some tips? Thanks~ The primary hyperparameters include Intensity Clipping Range, Global Foreground Voxel Intensity Mean/Std, Patch Size, and do_dummy_2D.

We are using the preprocessing given by nnU-Net (Intensity Clipping Range, Global Foreground Voxel Intensity Mean/Std, do_dummy_2D). You preprocess the data using nnUNET and it will output the schemes. You can also refer to its paper (nature.com/articles/s41592-020-01008-z) for details. Basically, nnU-Net uses the 0.5 and 99.5 percentiles of the foreground voxels for clipping as well as the global foreground mean and s.d. for the normalization of all images.

For patch size, we also look at the training data and calculate the size of the lesion, and select the size that can include most of the lesions in training data (even if the prompt is given in the margin of the lesion).

**Comment by 22TonyFStark** [comment]
@peterant1995 Thanks for your advice~
